src/main/scala/strawman/collection/View.scala

```diff
@@ -183,6 +184,12 @@ object View extends IterableFactory[View] {
       else -1
   }
 
+  /** A view that Computes the union between a set and another set.
+   */
+  case class Union[A](prefix: Iterable[A], suffix: Iterable[A]) extends View[A] {
+    def iterator() = prefix.iterator() ++ suffix.iterator()
+  }
```

Comment:
What’s the advantage over `Concat`?

Reply:
One reason is that the known size of a set union is not the same as the size of a concatenation (where it is always the sum of the sizes of the arguments) but for unions this does not hold since the size can be smaller if the same element exists in both arguments (so `knownSize` is not overridden in `Union` but instead inherited to be `-1`).

Reply:
Thanks for the explanation, good catch!